### PR TITLE
Add global configuration for SpellComponentsRequired

### PR DESCRIPTION
**Pull request by LtRipley36706**

### CODE REVIEW

**Comment by Mag-nus** [comment]
Didn't we already have a burn multiplier?

**Comment by LtRipley36706** [comment]
> Didn't we already have a burn multiplier?

not a comp burn change, this is a global override to force client/server to not require any spell components for casting

File: Source/ACE.Server/Managers/PropertyManager.cs [hunk 1]
```diff
@@ -502,6 +502,7 @@ public static void LoadDefaultProperties()
                 ("quest_info_enabled", new Property<bool>(false, "toggles the /myquests player command")),
                 ("salvage_handle_overages", new Property<bool>(false, "in retail, if 2 salvage bags were combined beyond 100 structure, the overages would be lost")),
                 ("show_dot_messages", new Property<bool>(false, "enabled, shows combat messages for DoT damage ticks. defaults to disabled, as in retail")),
+                ("spell_requires_spell_comps", new Property<bool>(true, "if FALSE players casting spells does not require spell components in inventory. defaults to enabled, as in retail")),
```
Comment (gmriggs) [inline Source/ACE.Server/Managers/PropertyManager.cs]: can this name be shortened?

require_spell_comps perhaps?

**Comment by Mag-nus** [comment]
> 
> 
> > Didn't we already have a burn multiplier?
> 
> not a comp burn change, this is a global override to force client/server to not require any spell components for casting

I thought you could change the burn multiplier to not consume any comps.

**Comment by LtRipley36706** [comment]
> > > Didn't we already have a burn multiplier?
> > 
> > 
> > not a comp burn change, this is a global override to force client/server to not require any spell components for casting
> 
> I thought you could change the burn multiplier to not consume any comps.

even if you can, that is not the same thing as this. This boolean makes so the client (and server) doesn't do any checking if your inventory contains comps

File: Source/ACE.Server/WorldObjects/Player_Magic.cs [hunk 2]
```diff
@@ -1230,6 +1231,8 @@ public bool HasComponentsForSpell(Spell spell)
         {            
             spell.Formula.GetPlayerFormula(this);
 
+            if (!PropertyManager.GetBool("spell_requires_spell_comps").Item) return true;
```
Comment (gmriggs) [inline Source/ACE.Server/WorldObjects/Player_Magic.cs]: get name does not match up with property name

File: Source/ACE.Server/WorldObjects/Player_Networking.cs [hunk 3]
```diff
@@ -103,6 +106,12 @@ private void SendSelf()
             SendContractTrackerTable();
         }
 
+        private void SendPropertyUpdatesAndOverrides()
+        {
+            if (!PropertyManager.GetBool("spell_requires_spell_comps").Item)
```
Comment (gmriggs) [inline Source/ACE.Server/WorldObjects/Player_Networking.cs]: get name does not match up with property names

**Pull request merged**